include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -264,8 +265,10 @@ struct transform_reduce
     operator()(const _NDItemId& __item_id, const _Size& __n, const _Size& __global_offset, const _AccLocal& __local_mem,
                const _Acc&... __acc) const
     {
-#ifndef _ONEDPL_DETECT_SPIRV_COMPILATION
-        if constexpr (_Commutative::value)
+        // For non-SPIRV targets, we check if the operator is commutative before selecting the appropriate codepath.
+        // On SPIRV targets, we always force the non-commutative implementation as this is more performant.
```

Comment:
```suggestion
        // On SPIRV targets, we always force the sequential implementation as this is more performant.
```

Lets try to talk about sequential vs non-sequential implementations, and non-commutative vs commutative operators, so that we do our best to not conflate one with the other.  

The non-commutative operator forces sequential implementation, but the either implementation is acceptable for commutative operators.

Reply:
Good point, I have added this change.